### ISSUE TITLE
im2col: partly uninitialized output from the second call on, when autotune picks a tile size smaller than the hard-coded launch grid assumes

### ISSUE THREAD

**Issue by truong-v** [FlagOpen/FlagGems#5224]
## Summary

`im2col_kernel` is autotuned over five `(BLOCK_M, BLOCK_N)` configs, but `_launch_im2col_kernel` sizes the launch grid with hard-coded tile sizes 64 and 128. With any winner except (64,128) the launch covers only part of the output — half the rows for `BLOCK_M=32`, half or a quarter of the columns for `BLOCK_N=64/32` — and the uncovered region of the `torch.empty` output is returned as uninitialized memory. Correct shape, no error.

The first call per shape always looks correct: the autotune sweep runs every candidate config against the same buffer, so it gets fully written no matter which config wins. Only from the second call on does the cached winner run alone — which is why the accuracy suite (one call per parameter set) passes, and why which shapes fail depends on which config wins the timing race on a given GPU.

The kernel was added in #3698.

## Reproduction

```python
import torch
import torch.nn.functional as F

import flag_gems

torch.manual_seed(0)
for shape in [(1, 4, 16, 16), (1, 8, 32, 32), (1, 64, 28, 28), (1, 32, 64, 64)]:
    x = torch.randn(shape, device="cuda")
    ref = F.unfold(x, kernel_size=3, padding=1)  # aten::im2col
    with flag_gems.use_gems():
        out1 = F.unfold(x, kernel_size=3, padding=1)  # 1st call: autotunes
        out2 = F.unfold(x, kernel_size=3, padding=1)  # 2nd call: cached config
    print(
        f"shape={shape}: max|gems - ref|  first call = "
        f"{(out1 - ref).abs().max().item():.3g}, "
        f"second call = {(out2 - ref).abs().max().item():.3g}"
    )
```

On an L40S:

```
shape=(1, 4, 16, 16): max|gems - ref|  first call = 0, second call = 3.32
shape=(1, 8, 32, 32): max|gems - ref|  first call = 0, second call = 4
shape=(1, 64, 28, 28): max|gems - ref|  first call = 0, second call = 4.17
shape=(1, 32, 64, 64): max|gems - ref|  first call = 0, second call = 4.41
```

On an H100 NVL 3 of the 4 shapes fail — the autotuner happens to pick the covering config for `(1, 8, 32, 32)` there. Pinning the tuning table to a single one of its own configs makes the failure deterministic from the first call.

(Open PR #4727 bundles an equivalent grid change inside the `im2col_out` promotion, but the corruption itself is not tracked anywhere and no test guards it, so filing it for the operator as shipped today.)

## Environment

- FlagGems master (75cf3ea)
- NVIDIA L40S and H100 NVL
- torch 2.6.0+cu124, triton 3.2.0
